### DISCUSSION TITLE
lsp-zero cannot be loaded in Lazy

### DISCUSSION THREAD

**Discussion by PrathmeshVishwakarma** [VonHeikemen/lsp-zero.nvim#400] [General]
Hello, just a quick question, new user here. So I read the guide and it says to add the snippet at the end of the file
here it is:
![image](https://github.com/VonHeikemen/lsp-zero.nvim/assets/170882272/6b6a1c30-9f21-49b9-9905-61e53064ea61)
but when I do, it says lsp-zero failed to load.
I initially put the config in the same plugin folder too, but it too didn't load lsp-zero (but it did install lsp-zero, mason, ..... etc)
Please Help. Thanks for your time

**Comment by PrathmeshVishwakarma** [comment]
And, in the guide lazy loading with lazy nvim, the config is inside the return function of lazy
as in
```
return {
".../lsp-zero.nvim"
config = function()
.....
}
```

**Reply by VonHeikemen** [reply]
The snippet in the lazy load guide assumes you are going to setup lsp-zero in its own dedicated file, that's why there is a return statement.

If you are new to the many ways you can setup plugins in lazy.nvim I recommend you read this blogpost: [Lazy.nvim: plugin configuration](https://dev.to/vonheikemen/lazynvim-plugin-configuration-3opi)

**Comment by VonHeikemen** [comment]
What's the error message you get?

At that point of the tutorial your `init.lua` should look like this.

```lua
local lazypath = vim.fn.stdpath('data') .. '/lazy/lazy.nvim'
local uv = vim.uv or vim.loop

-- Auto-install lazy.nvim if not present
if not uv.fs_stat(lazypath) then
  print('Installing lazy.nvim....')
  vim.fn.system({
    'git',
    'clone',
    '--filter=blob:none',
    'https://github.com/folke/lazy.nvim.git',
    '--branch=stable', -- latest stable release
    lazypath,
  })
  print('Done.')
end

vim.opt.rtp:prepend(lazypath)

require('lazy').setup({
  {'folke/tokyonight.nvim'},
  {'VonHeikemen/lsp-zero.nvim', branch = 'v3.x'},
  {'neovim/nvim-lspconfig'},
  {'hrsh7th/cmp-nvim-lsp'},
  {'hrsh7th/nvim-cmp'},
  {'L3MON4D3/LuaSnip'},
})

vim.opt.termguicolors = true
vim.cmd.colorscheme('tokyonight')

local lsp_zero = require('lsp-zero')

lsp_zero.on_attach(function(client, bufnr)
  -- see :help lsp-zero-keybindings
  -- to learn the available actions
  lsp_zero.default_keymaps({buffer = bufnr})
end)
```

**Comment by PrathmeshVishwakarma** [comment]
> The snippet in the lazy load guide assumes you are going to setup lsp-zero in its own dedicated file, that's why there is a return statement.

Oh, I see whats the issue here, I was doing the above on my seperate lsp-zero.lua file. Thanks for pointing out and helping.
Cheers! have a good day
